app/models/internal_post.rb

```diff
@@ -0,0 +1,56 @@
+class InternalPost < Post
+  belongs_to :user
+  validates :user, :title, :body_markdown, presence: true
+  has_many :comments, as: :parent
+
+  before_save do
+    parser = Redcarpet::Markdown.new(Redcarpet::Render::HTML, autolink: true)
+    self.body_html = parser.render(body_markdown)
+  end
+
+  after_create :notify_twitter
```

Comment:
External Posts should also notify twitter. Please move this back to the `Post` class.